### ISSUE TITLE
Can't switch back to primary monitor

### ISSUE THREAD

**Issue by jriker1** [Nonary/MonitorSwapAutomation#6]
Using Sunshine .20 in Windows with Windows client accessing the desktop.  When I use Moonlight I am able to access the dummy plug but when I disconnect it keeps failing to connect to the primary monitor.  Dummy is connected to the HDMI port on the RTX 4090 and the primary display is connected to the top DisplayPort on the RTX 4090.
[log.txt](https://github.com/Nonary/MonitorSwapAutomation/files/11697418/log.txt)
[dummy.cfg.txt](https://github.com/Nonary/MonitorSwapAutomation/files/11697435/dummy.cfg.txt)
[primary.cfg.txt](https://github.com/Nonary/MonitorSwapAutomation/files/11697436/primary.cfg.txt)
[settings.json.txt](https://github.com/Nonary/MonitorSwapAutomation/files/11697437/settings.json.txt)

**Comment by Nonary** [comment]
Some displays fail to set back until you’ve come back to the PC and woken up to display, is it failing to swap back when you do this still?

**Comment by jriker1** [comment]
Yes I thought my keyboard and mouse were bad because the screen wasn't coming on.  I can type on the keyboard and then when I moonlight in can see what I was typing but the monitor never comes on.

**Comment by Nonary** [comment]
> ////Yes I thought my keyboard and mouse were bad because the screen wasn't coming on. I can type on the keyboard and then when I moonlight in can see what I was typing but the monitor never comes on. When I move the mouse or tap on the keyboard or hit the windows key nothing. In the logs the below continues over and over /Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen Attempting to set primary screen
> 
> Note screen itself is sleeping so to speak. Green light with the yellow sleeping light. Never turns green when moving around the mouse or keyboard.
> 
> if I RDP in and exit nicely with a batch file as recommended:
> 
> ```
> for /f "usebackq skip=1 tokens=3" %%i in (`query user %USERNAME%`) do %windir%\System32\tscon.exe %%i /dest:console
> exit 
> ```
> 
> still doesn't come back or if I RDP in and just badly drop doesn't come back. Which then I can't get back in thru moonlight.
> 
> If I then fix it by RDP in and exiting with the batch file I then moonlight in and manually switch the screen to extend the desktop. The monitor then turns on.
> 
> Sorry keep troubleshooting.
> 
> Maybe expected but when I'm moonlight in with just the dongle active and run dxgi-info it shows \.\DISPLAY1 with 4k res and also Microsoft Basic Render Driver. Doesn't list the other physical monitor but maybe normal.
> 
> If I manually disable just enable DISPLAY2 the primary Monitor becomes active but dxgi-info still shows it as \.\DISPLAY1. In the display screen in windows shows it as display "2".
> 
> Do the both become \.\DISPLAY1 after the other is disabled?

I wouldn’t worry about the display names, they change randomly due to windows bugs. That’s why the tool loads a profile instead as it is a workaround.


The script is unable to set the screen if you’ve disconnected using RDP due to how windows detaches the UI so to speak when that is done.

the script also cannot set the screen from a reboot or logouts, again due to limitations of windows. Basically monitor settings are user specific and it requires user to be signed in.

**Comment by Nonary** [comment]
Resolved on discord.

Basically, we had you use Moonlight remotely. First, you configured your PC to use the dummy plug, and then you exported it in the multi monitor tool as dummy.cfg. After that, you switched back to your primary screen manually, then exported it again as "primary.cfg". Instead of manually zeroing out things, we exported the profiles after manually changing the screens, and that seemed to work.
